Jenkinsfile

```diff
@@ -95,20 +95,30 @@ def rpm_test_daos_test = '''me=\\\$(whoami)
                             trap 'set -x; kill -INT \\\$AGENT_PID \\\$COPROC_PID' EXIT
                             OFI_INTERFACE=eth0 daos_test -m'''
 
-def rpm_scan_daos_test = '''pushd /tmp
-                              wget \
-                              http://rfxn.com/downloads/maldetect-current.tar.gz
-                              rm -rf /tmp/maldet-current
-                              mkdir -p /tmp/maldet-current
-                              tar -C /tmp/maldet-current --strip-components=1 \
-                                -xf maldetect-current.tar.gz
-                              cd maldet-current
+def get_lmd_script = '''set -ex
+                        lmd_tarball='maldetect-current.tar.gz'
+                        lmd_src='maldet-current'
+                        rm -rf ./${lmd_src}
+                        mkdir -p ${lmd_src}
+                        if test -e "${lmd_tarball}"; then
+                          zflag="-z ${lmd_tarball}"
+                        else
+                          zflag=
+                        fi
+                        curl http://rfxn.com/downloads/${tarball} \
+                          ${zflag} --silent --show-error --fail \
+                          -o ${tarball}
+                        tar -C ${lmd_src} --strip-components=1 \
+                          xf ${lmd_tarball}'''
```

Comment:
You could avoid even having to untar in the case that there is not a newer file downloaded by testing if the tarball is different using a hashing tool like `md5sum`.  Take the sum before and after the `curl` and if they are the same, skip the `rm` and `untar`.

Reply:
Added to current pushed commit.